board/board.h

```diff
@@ -7,39 +7,53 @@
 // ///// Board definition and detection ///// //
 #include "drivers/harness.h"
 #ifdef PANDA
-  #include "drivers/fan.h"
-  #include "drivers/rtc.h"
-  #include "drivers/clock_source.h"
-  #include "boards/white.h"
-  #include "boards/grey.h"
-  #include "boards/black.h"
-  #include "boards/uno.h"
-  #include "boards/dos.h"
+  #ifdef STM32H7
+    #include "boards/red.h"
+#include "drivers/rtc.h"
+  #else
+    #include "drivers/fan.h"
+    #include "drivers/rtc.h"
+    #include "drivers/clock_source.h"
+    #include "boards/white.h"
+    #include "boards/grey.h"
+    #include "boards/black.h"
+    #include "boards/uno.h"
+    #include "boards/dos.h"
+  #endif
 #else
   #include "boards/pedal.h"
 #endif
 
 void detect_board_type(void) {
   #ifdef PANDA
+    #ifdef STM32H7
+      if(!detect_with_pull(GPIOF, 7, PULL_UP) && !detect_with_pull(GPIOF, 8, PULL_UP) && !detect_with_pull(GPIOF, 9, PULL_UP) && !detect_with_pull(GPIOF, 10, PULL_UP)){
```

Comment:
For the H7 line with the new board detection pins, let's already abstract this in a function to read the ID from those pins, and return it as a uint8